tests/scripts/thread-cert/node.py

```diff
@@ -4145,6 +4145,34 @@ def start_pd_radvd_service(self, prefix):
 """ % (prefix,))
         self._start_radvd_and_verify()
 
+    def start_rdnss_radvd_service(self, rdnss):
+        """
+        Start radvd service to advertise RDNSS.
+
+        Args:
+            rdnss (str): The IPv6 address of the recursive DNS server to advertise.
+        """
+        self.bash("""cat >/etc/radvd.conf <<EOF
+interface eth0
+{
+    AdvSendAdvert on;
+
+    AdvReachableTime 20;
+    AdvRetransTimer 20;
+    AdvDefaultLifetime 180;
+    MinRtrAdvInterval 120;
+    MaxRtrAdvInterval 180;
+    AdvDefaultPreference low;
+                  
+    RDNSS %s
+    {
+        AdvRDNSSLifetime 1800;
+    };
+};
+EOF
+""" % (rdnss,))
```

Comment:
IIRC the `%` format string is no longer suggested. Maybe you can use `f-` string or `str.format()`.